MekHQ/src/mekhq/gui/dialog/CampaignOptionsDialog.java

```diff
@@ -3327,9 +3327,37 @@ public void actionPerformed(ActionEvent e) {
         panRandomPortrait.setName("panRandomPortrait"); // NOI18N
         panRandomPortrait.setLayout(new BorderLayout());
 
-        JPanel panUsePortrait = new JPanel(new GridLayout(nRow, 4));
-
+        JPanel panUsePortrait = new JPanel(new GridLayout((int) Math.ceil((Person.T_NUM + 2) / 4.0), 4));
```

Comment:
This is begging for a comment to explain what all that math is for.

Reply:
Added